tests/nodetree/test_node_unit.py

```diff
@@ -21,12 +22,264 @@
     MockLeafNode,
     MockMetaNode,
     MockNode,
+    MockNodeTreeManager,
     MockPartialNode,
     MockResultNode,
     MockWildcardNode,
 )
 
 
+class TestNodeInfo:
+    def test_getattr(self):
+        info = NodeInfo({"Properties": "p"})
+        assert info.Properties == "p"
+        assert info.properties == "p"
+
+    def test_contains(self):
+        with patch.object(
+            NodeInfo,
+            "__dir__",
+            autospec=True,
+            return_value=["a"],
+        ) as dir_mock:
+            info = NodeInfo({})
+            assert "a" in info
+            dir_mock.assert_called_once_with(info)
```

Comment:
issue: This test does not test our API but python itself. Better use something like 

```suggestion
            info = NodeInfo({"a" : "Test"})
            assert "a" in info
```

Reply:
actually, the `in` operator calls the `__contains__` and we want to test that it calls `__dir__` , which is not trivial. I think the proposed changes make the test less of a unittest